File: product-service/src/main/java/com/intexsoft/webshop/productservice/dto/attributevalue/AttributeValueUpdateDto.java
package com.intexsoft.webshop.productservice.dto.attributevalue;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.annotation.Nullable;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.Size;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@JsonIgnoreProperties(ignoreUnknown = true)
public class AttributeValueUpdateDto {
    @NotNull
    @Positive
    Long attributeId;
    @Nullable
    @Pattern(regexp = "^(?!\\s*$).+")
    @Size(max = 50)
    String value;
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/dto/attributevalue/AttributeValueUpdateDto.kt
package com.intexsoft.webshop.productservicekt.dto.attributevalue

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Positive
import jakarta.validation.constraints.Size

@JsonIgnoreProperties(ignoreUnknown = true)
data class AttributeValueUpdateDto(
    @field:Positive
    val attributeId: Long,
    @field:Pattern(regexp = "^(?!\\s*$).+")
    @field:Size(max = 50)
    val value: String?
)